src/decoder/serde.rs

```diff
@@ -689,6 +689,18 @@ impl<'de> Deserialize<'de> for TimeStamp {
     }
 }
 
+impl<'de> Deserialize<'de> for RegExp {
```

Comment:
This follows the pattern of the other deserializers defined here. Since the deserialize implementation for `Bson` implements deserializing from extjson, we piggyback off that to either get a RegExp or return an error.